pkg/cli/cmd/pipelinerun/list.go

```diff
@@ -0,0 +1,261 @@
+	"github.com/tektoncd/cli/pkg/cli"
+	"github.com/tektoncd/cli/pkg/formatted"
+	v1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1"
+	"github.com/tektoncd/results/pkg/cli/common"
+	"github.com/tektoncd/results/pkg/cli/config"
+	pb "github.com/tektoncd/results/proto/v1alpha2/results_go_proto"
+)
+
+const prListTemplate = `{{- $prl := len .PipelineRuns.Items -}}{{- if eq $prl 0 -}}
+No PipelineRuns found
+{{ else -}}
+{{- if not $.NoHeaders -}}
+{{- if $.AllNamespaces -}}
+NAMESPACE	NAME	UID	STARTED	DURATION	STATUS
+{{ else -}}
+NAME	UID	STARTED	DURATION	STATUS
+{{ end -}}
+{{- end -}}
+{{- range $_, $pr := .PipelineRuns.Items }}{{- if $pr }}{{- if $.AllNamespaces -}}
+{{ $pr.Namespace }}	{{ $pr.Name }} {{ $pr.UID }}	{{ formatAge $pr.Status.StartTime $.Time }}	{{ formatDuration $pr.Status.StartTime $pr.Status.CompletionTime }}	{{ formatCondition $pr.Status.Conditions }}
+{{ else -}}
+{{ $pr.Name }}	{{ $pr.UID }}	{{ formatAge $pr.Status.StartTime $.Time }}	{{ formatDuration $pr.Status.StartTime $pr.Status.CompletionTime }}	{{ formatCondition $pr.Status.Conditions }}
+{{ end -}}{{- end -}}{{- end -}}
+{{- end -}}`
+
+type listOptions struct {
+	Client        *client.RESTClient
+	Limit         int32
+	AllNamespaces bool
+	Label         string
+	SinglePage    bool
+	PipelineName  string
+}
+
+// listCommand initializes a cobra command to list PipelineRuns
+func listCommand(p common.Params) *cobra.Command {
+	opts := &listOptions{
+		Limit:         10,
+		AllNamespaces: false,
+		SinglePage:    false,
+	}
+
+	eg := `List all PipelineRuns in a namespace 'foo':
+    tkn-results pipelinerun list -n foo
+
+List all PipelineRuns in 'default' namespace:
+    tkn-results pipelinerun list -n default
+
+List PipelineRuns with a specific label:
+    tkn-results pipelinerun list -l app=myapp
+
+List PipelineRuns from all namespaces:
+    tkn-results pipelinerun list -A
+
+List PipelineRuns with limit of 20 per page:
+    tkn-results pipelinerun list --limit=20
+
+List PipelineRuns for a specific pipeline:
+    tkn-results pipelinerun list foo -n namespace
+
+List PipelineRuns with partial pipeline name match:
+    tkn-results pipelinerun list build -n namespace
+`
+	cmd := &cobra.Command{
+		Use:     "list [pipeline-name]",
+		Aliases: []string{"ls"},
+		Short:   "List PipelineRuns in a namespace",
+		Annotations: map[string]string{
+			"commandType": "main",
+		},
+		Example: eg,
+		PreRunE: func(_ *cobra.Command, args []string) error {
+			c, err := config.NewConfig(p)
+			if err != nil {
+				return err
+			}
+			opts.Client, err = client.NewRESTClient(c.Get())
+			if err != nil {
+				return err
+			}
+
+			if opts.Limit < 5 || opts.Limit > 100 {
+				return errors.New("limit should be between 5 and 100")
+			}
+
+			if len(args) > 0 {
+				opts.PipelineName = args[0]
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			// Build filter string
+			filter := []string{`data_type==PIPELINE_RUN`}
+			if opts.Label != "" {
```

Comment:
Should we have some validation here for label format before making request to results?

Reply:
@khrm please take a look, did the required changes.